tf2onnx/optimizer/merge_duplicated_nodes_optimizer.py

```diff
@@ -90,3 +92,41 @@ def _skip_node_type(node):
     @staticmethod
     def _len_of_node_output(node):
         return len(node.output)
+
+    @staticmethod
+    def _are_attr_equal(a, b):
+        if len(a) != len(b):
+            return False
+        for k, v in a.items():
+            if v.type == onnx_pb.AttributeProto.FLOAT or \
+                    v.type == onnx_pb.AttributeProto.INT or \
+                    v.type == onnx_pb.AttributeProto.STRING or \
+                    v.type == onnx_pb.AttributeProto.FLOATS or \
+                    v.type == onnx_pb.AttributeProto.INTS or \
+                    v.type == onnx_pb.AttributeProto.STRINGS:
+                if v != b[k]:
+                    return False
+            elif v.type == onnx_pb.AttributeProto.TENSOR:
+                t1 = v.t
+                t2 = b[k].t
+                if not (t1.dims == t2.dims and t1.data_type == t2.data_type and t1.raw_data == t2.raw_data):
+                    return False
+            elif v.type == onnx_pb.AttributeProto.TENSORS:
+                if len(v.ts) != len(b[k].ts):
+                    return False
+                for i in range(len(v.ts)):
+                    t1 = v.ts[i]
+                    t2 = b[k].ts[i]
+                    if not (t1.dims == t2.dims and t1.data_type == t2.data_type and t1.raw_data == t2.raw_data):
+                        return False
+            elif v.type == onnx_pb.AttributeProto.GRAPH:
+                # TODO provide implementation which skips doc_string and name
+                # TODO We need an equals() method for each Node and every Proto object
+                # TODO Using that graphs (and their nodes) could be compared recursively
+                if not v == b[k]:
+                    return False
+            elif v.type == onnx_pb.AttributeProto.GRAPHS:
+                # TODO provide implementation which skips doc_string and name
+                if not v == b[k]:
+                    return False
+        return True
```

Comment:
reture false may be better? for we are not sure we have cover all attribute type.

Reply:
I'm going to rewrite this section so it's a little more robust